frontend_vue/src/components/tokens/types.ts

```diff
@@ -107,38 +107,59 @@ type genericObjectType = {
   [k: string]: string;
 };
 
+export type GeoInfoType = {
+  loc: string;
+  org: string;
+  city: string;
+  country: string;
+  region: string;
+  hostname: string;
+  ip: string;
+  timezone: string;
+  postal: string;
+  asn: {
+    route: string;
+    type: string;
+    asn: string;
+    domain: string;
+    name: string;
+  };
+  readme: string;
+};
+
+export type RequestHeadersType = {
+  Host: string;
+  'X-Real-Ip': string;
+  'X-Forwarded-For': string;
+  'X-Forwarded-Host': string;
+  Connection: string;
+  'Upgrade-Insecure-Requests': string;
+  'User-Agent': string;
+  Accept: string;
+  'Accept-Encoding': string;
+  'Accept-Language': string;
+};
```

Comment:
i suspect there are going to be lots of other headers here. can we make this just an `Any` dict? What happens when we get headers we don't expect? Breakage?

Reply:
Yeah we can make it all generic, like I did for additional_info and request args
`export type HitsType = {
  time_of_hit: number;
  src_ip: string;
  geo_info: GeoInfoType;
  is_tor_relay: boolean;
  input_channel: string;
  src_data: genericObjectType;
  useragent: string;
  token_type: string;
  request_headers: RequestHeadersType;
  additional_info: {
    [k: string]: {
      [k: string]: string[];
    };
  };
  request_args: {
    [k: string]: {
      [k: string]: string[];
    };
  };
};`

Nothing happens if the API returns something different, Typescript checks only at compile time

Reply:
Yeah i'm not sure it's useful keeping a list of this then as the headers object is going to change depending on the requests